### PR TITLE
enhancement(elixir): Handle user leaves

### LINKED ISSUE #40: Handle channel quitting on location change

### PR DESCRIPTION
**Pull request by mrapacz**
**Summary**

giphy 3

This PR introduces [Phoenix.Presence](https://hexdocs.pm/phoenix/Phoenix.Presence.html) to track  users who are present in a room. It also introduces changes mentioned in #40 - whenever you join a room and change your location, the change of game state should be registered.

The PR is not perfect, Presence is not sending out information when people are leaving the room empty. We will need to add a tool responsible for monitoring and handling that, but it shall be part of #85.

I have also moved out parts of the socket handling in Elm to *Socket.elm* because the *Update.elm* was getting really awful. 

**Related issues**

closes: #40 , #85, #89 

**Test plan**
Test it manually:
I:
1. Join a room using two accounts
2. Close browser tab responsible for one of the two players' game
3. See the updated list of players (user left)

giphy 4

II:
1. Join a room using two accounts
2. Go back to the room lobby as one of the players
3. See the updates list of players (user left)

giphy 5

### CODE REVIEW

**Comment by mrapacz** [comment]
@pmrukot 
Pushed improved version. Now *all* the leave cases are covered. All except for #89, which is going to be a part of different PR once we decide about it. 

As for now, whenever a room is left empty, the genserver responsible for it is getting stopped. Try it out.

**Comment by mrapacz** [comment]
@pmrukot Handling the page refresh while in game room turned out to be pretty easy. The user is redirected to the start page with this implementation.

File: aion/web/elm/src/Socket.elm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    Phoenix.Socket.leave ("rooms:" ++ roomId) socket
+
+
+
+--
+-- let
+--     push_ =
+--         Phoenix.Push.init "user:left" ("rooms:" ++ roomId)
+-- in
+--     Phoenix.Socket.push push_ socket
```
Comment (pmrukot) [inline aion/web/elm/src/Socket.elm]: Do we want this commented out code to be merged?

**Review by pmrukot — APPROVED** [review]
Looks great, 👍 for extracting socket logic to separate file.

File: aion/web/elm/src/Socket.elm (hunk 1 above)
Comment (mrapacz) [reply aion/web/elm/src/Socket.elm]: Nice catch.

**Comment by mrapacz** [comment]
@pmrukot going to merge this PR as soon as CI goes green. Thanks for the review, you da real MVP.

1uxx14

**Pull request merged**